File: flutter_version/lib/src/features/news/domain/entities/articles_page.dart
import 'package:equatable/equatable.dart';
import 'article.dart';

class ArticlesPage extends Equatable {
  final List<Article> articles;
  final int totalResults;

  const ArticlesPage({required this.articles, required this.totalResults});

  @override
  List<Object?> get props => [articles, totalResults];
}

File: swiftui_version/News/News/Features/News/Domain/Entities/ArticlesPage.swift
import Foundation

/// Domain entity representing a paginated response of articles
struct ArticlesPage: Equatable {
    let articles: [Article]
    let totalResults: Int
    
    static func == (lhs: ArticlesPage, rhs: ArticlesPage) -> Bool {
        lhs.articles == rhs.articles && lhs.totalResults == rhs.totalResults
    }
}
